thirdparty/minizip/CMakeLists.txt

```diff
@@ -13,14 +13,23 @@ assert_var_defined(AR)
 assert_var_defined(LDFLAGS)
 assert_var_defined(RANLIB)
 
+if ($ENV{DARWIN})
+    #fix build error due to implicit declarations of function being invalid in C99 under macOS/XCode 12
+    set(CFLAGS "${CFLAGS} -Wno-error=implicit-function-declaration")
+endif()
+
 ep_get_source_dir(SOURCE_DIR)
 
 # NOTE: 53a657318af1fccc4bac7ed230729302b2391d1d is the tip of the 1.2 branch. The fcrypt API we need is gone in master.
 # FIXME: Even then, something in said branch seems to upset MuPDF with our custom patch as-is, so keep using the old code...
 #set(PATCH_CMD "${KO_PATCH} ${CMAKE_CURRENT_SOURCE_DIR}/minizip-libaes-makefile.patch")
 
 set(PATCH_CMD sh -c "${ISED} \"s|^CC=|#CC=|g\" aes/Makefile && ${ISED} \"s|^CFLAGS=|#CFLAGS=|g\" aes/Makefile")
-set(BUILD_CMD_STR -j${PARALLEL_JOBS} -C aes CC=${CC} LDFLAGS=${LDFLAGS} AR=${AR} RANLIB=${RANLIB})
+if ($ENV{DARWIN})
+    set(BUILD_CMD_STR -j${PARALLEL_JOBS} -C aes CC=${CC} CFLAGS=${CFLAGS} LDFLAGS=${LDFLAGS} AR=${AR} RANLIB=${RANLIB})
+else()
+    set(BUILD_CMD_STR -j${PARALLEL_JOBS} -C aes CC=${CC} LDFLAGS=${LDFLAGS} AR=${AR} RANLIB=${RANLIB})
+endif()
```

Comment:
You can probably drop the DARWIN branch and simply always pass CFLAGS, it's probably what it should already have been doing anyway ;).

Reply:
I certainly don't mind doing that. Are there any concerns about breaking the build on other platforms? I've only built this for the Mac desktop.

Reply:
Btw, you should also add it on line 12 or so in that case, and I imagine it's not passed on from Makefile.third either? But agreed it should probably be passed anyway.

Reply:
I've removed the DARWIN checks. Is it still necessary to add at line 12 or look at Makefile.third?

Reply:
Looking at it, it appears to go through a dirty, dirty hack: it's passing `CFLAGS` via `CC`. That may be on purpose (i.e., the Makefiles are broken), in which case, you probably don't even need the `BUILD_CMD_STR` bit...

Reply:
Yep, the Makefile doesn't honor CFLAGS.

TL;DR:

* You don't have to tweak our Makefile.third
* You can also stop passing CFLAGS to BUILD_CMD_STR

I have no idea if updating CFLAGS in CMake will magically be reflected in CC, but, oh, well.

Reply:
@JasonInOttawa Comments on specific lines are just that! We were just talking about this one specific if DARWIN, not all of them. ;-)

Reply:
Thank you. Revising BUILD_CMD_STR from this

`set(BUILD_CMD_STR -j${PARALLEL_JOBS} -C aes CC=${CC} CFLAGS=${CFLAGS} LDFLAGS=${LDFLAGS} AR=${AR} RANLIB=${RANLIB})
`
to this

`set(BUILD_CMD_STR -j${PARALLEL_JOBS} -C aes CC=${CC} LDFLAGS=${LDFLAGS} AR=${AR} RANLIB=${RANLIB})
`

results in the implicit function declaration error. I apologize if I misunderstood the comment, but if I did understand then passing CFLAGS in BUILD_CMD_STR is required for the build to work.

Reply:
You'll want to tweak `CC` (which already includes CFLAGS), c.f., https://github.com/koreader/koreader-base/blob/7231add42900cb5879e1e0e7d1c98aec4f71112c/Makefile.third#L108

Reply:
By "tweak CC" do you mean add the flag to the CC variable before setting BUILD_CMD_STR? Or should this change be done in Makefile.third or Makefile.def? Or am I chasing completely the wrong cat?

```
set(CC "${CC} -Wno-error=implicit-function-declaration")
set(BUILD_CMD_STR -j${PARALLEL_JOBS} -C aes CC=${CC} LDFLAGS=${LDFLAGS} AR=${AR} RANLIB=${RANLIB})
```

Thank you!

Reply:
Yep, that should work ;).

Reply:
The change is in. Thank you for your help!